examples/multimodal/components/processor.py

```
@@ -263,8 +389,86 @@ async def generate(self, raw_request: MultiModalRequest):
             # reconstructing back the OpenAI chat response as dynamo egress expects it
             if response.startswith("data: [DONE]"):
                 break
-            response = json.loads(response.lstrip("data: "))
-            yield response
+            
+            # Handle both streaming (with "data: " prefix) and non-streaming responses
+            if response.startswith("data: "):
+                response = json.loads(response.lstrip("data: "))
```

Ruff (0.13.3)

Using `.strip()` with multi-character strings is misleading

(B005)